hw/ip/keymgr/rtl/keymgr_ctrl.sv

```
@@ -193,7 +205,13 @@ module keymgr_ctrl import keymgr_pkg::*;(
         // the values here
         else begin
           cnt_clr = 1'b1;
-          key_state_d[0] = key_state_q[0] ^ root_key_i;
+
+          // absorb key if valid, otherwise use entropy
+          if (root_key_valid_q) begin
+            key_state_d[0] = root_key_i.key_share0;
+            key_state_d[1] = root_key_i.key_share1;
+          end
```

@tjaychen 
sorry for commenting after merge
this root_key from OTP comes with a valid signal. When may OTP drive this valid to 0?

We don't document this valid signal in keymgr doc. The other data from OTP, flash doesn't come with valid signal.
Now we just silently use entropy and move to next state with successful status when valid=0. Should we notify the SW in this case?

---

o the valid is actually tied to whether this value is programmed in the OTP.
Regarding whether we should notify software... that's a good point. It's probably something we should add to let them know we are operating on entropy instead.

---

For the other data like flash_i.seeds and otp_i.devid, they don't have valid. is it safe to assume those data are ready when keymgr uses them?

---

yes it's okay to assume that. 
It's actually okay to assume that for the keys also.  The valid signal is not really there to mean the data may become ready later, but just tells the key manager to not take the value if it has never been set up.

---

I see. Then maybe let the SW know entropy is used for root key as you mentioned.